clash-prelude/src/Clash/Signal/Internal.hs

```diff
@@ -93,104 +132,517 @@ where
+-- be used in combination with 'createDomain'. For example, if you just want to
+-- change the period but leave all other settings in tact use:
+--
+-- > createDomain vSystem{vTag="System10", vPeriod=10}
+--
+vSystem :: VDomainConfiguration
+vSystem = vDomain (knownDomain @System)
+
+-- | A clock (and reset) dom with clocks running at 100 MHz. Memory elements
+-- respond to the rising edge of the clock, and asynchronously to changes in
+-- reset signals. It has defined initial values, and active-high resets.
+--
+-- See module documentation of "Clash.Explicit.Signal" for more information on
+-- how to create custom synthesis domains.
+type System = "System"
+
+
+-- | Convenience value to allow easy "subclassing" of IntelSystem domain. Should
+-- be used in combination with 'createDomain'. For example, if you just want to
+-- change the period but leave all other settings in tact use:
+--
+-- > createDomain vIntelSystem{vTag="Intel10", vPeriod=10}
+--
+vIntelSystem :: VDomainConfiguration
+vIntelSystem = vDomain (knownDomain @IntelSystem)
+
+-- | A clock (and reset) dom with clocks running at 100 MHz. Memory elements
+-- respond to the rising edge of the clock, and asynchronously to changes in
+-- reset signals. It has defined initial values, and active-high resets.
+--
+-- See module documentation of "Clash.Explicit.Signal" for more information on
+-- how to create custom synthesis domains.
+type IntelSystem = "IntelSystem"
+
+-- | Convenience value to allow easy "subclassing" of XilinxSystem domain. Should
+-- be used in combination with 'createDomain'. For example, if you just want to
+-- change the period but leave all other settings in tact use:
+--
+-- > createDomain vXilinxSystem{vTag="Xilinx10", vPeriod=10}
+--
+vXilinxSystem :: VDomainConfiguration
+vXilinxSystem = vDomain (knownDomain @XilinxSystem)
+
+-- | A clock (and reset) dom with clocks running at 100 MHz. Memory elements
+-- respond to the rising edge of the clock, and synchronously to changes in
+-- reset signals. It has defined initial values, and active-high resets.
+--
+-- See module documentation of "Clash.Explicit.Signal" for more information on
+-- how to create custom synthesis domains.
+type XilinxSystem = "XilinxSystem"
+
+-- | Same as SDomainConfiguration but allows for easy updates through record update syntax.
+-- Should be used in combination with 'vDomainConfiguration' and 'createDomain'. Example:
+--
+-- > createDomain (knownVDomain @System){vTag="System10", vPeriod=10}
+--
+-- This duplicates the settings in the "System" domain, replaces the name and
+-- period, and creates an instance for it. As most users often want to update
+-- the system domain, a shortcut is available in the form:
+--
+-- > createDomain vSystem{vTag="System10", vPeriod=10}
+--
+data VDomainConfiguration
+  = VDomainConfiguration
+  { vTag    :: String
+  -- ^ Corresponds to '_name' on 'DomainConfiguration'
+  , vPeriod :: Integer
```

Comment:
Any reason for using `Integer` rather than `Natural`?

Reply:
No! Thanks.